source/vstd/set_lib.rs

```diff
@@ -950,10 +950,10 @@ pub proof fn lemma_subset_equality<A>(x: Set<A>, y: Set<A>)
 
 /// If an injective function is applied to each element of a set to construct
 /// another set, the two sets have the same size.
-// the dafny original lemma reasons with partial function f
 pub proof fn lemma_map_size<A, B>(x: Set<A>, y: Set<B>, f: spec_fn(A) -> B)
     requires
-        injective(f),
+        forall|a1: A, a2: A|
```

Comment:
This seems useful enough that I think it's worth factoring into a spec function, e.g.,
`spec fn injective_on<X,Y>(r: spec_fn(X) -> Y, s: Set<X>) -> bool` in `relations.rs`.